plugins/modules/ndo_schema_template_deploy.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Akini Ross (@akinross) <akinross@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: ndo_schema_template_deploy
+short_description: Deploy schema templates to sites
+description:
+- Deploy schema templates to sites.
+author:
+- Akini Ross (@akinross)
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template.
+    type: str
+    required: yes
+  sites:
+    description:
+    - The name of the site(s).
+    type: list
+    elements: str
+  state:
+    description:
+    - Use C(deploy) to deploy schema template.
+    - Use C(redeploy) to redeploy schema template.
+    - Use C(undeploy) to deploy schema template from a site.
+    - Use C(query) to get deployment status.
+    type: str
+    choices: [ deploy, redeploy, undeploy, query ]
```

Comment:
Which option is used to delete the schema template?

Reply:
you don't delete a template but undeploy it from a site thus the state differ from other modules